ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -466,7 +672,131 @@ public void replicate() {
+   *    1. Exchange metadata : fetch the metadata of blobs added to remote replica since the last synchronization point
+   *    and filter the ones missing in local store.
+   *    2. Fetch missing blobs: fetch the missing blobs by issuing GET request to remote replica and write them to
+   *       the local store
+   *
+   *  During cross-colo replication, depending on the {@link ReplicationModelType}, the missing blobs are either fetched
+   *  from all remote replicas (if modelType == ALL_TO_ALL) or only fetched for local leader replicas from their remote
+   *  leader replicas (if modelType == LEADER_BASED). In the latter case, non-leader replica pairs (leader <-> standby,
+   *  standby <-> leader, standby <-> standby) will get their missing blobs from their corresponding leader<->leader
+   *  exchanges and intra-dc replication.
+   *
+   *  Here is a table listing on what is exchanged between local and remote replicas based on their roles
+   *  (leader/standby) when {@link ReplicationModelType is LEADER_BASED}.
+   *
+   *              |   Local Leader    |     Local Standby   |   Remote Leader   |  Remote Standby
+   *            -------------------------------------------------------------------------------------
+   *     Leader:  |        ---        |  metadata and data  | metadata and data |   metadata only
+   *     Standby: | metadata and data |  metadata and data  | metadata only     |   metadata only
+   *
+   */
+  public void replicateAcyclic() {
+    long oneRoundStartTimeMs =  time.milliseconds();
+    shouldTerminateCurrentCycle = false;
+
+    logger.trace("Thread name: {} Start RemoteReplicaGroup replication", threadName);
+
+    exchangeMetadataResponsesInEachCycle = new HashMap<>();
+    Set<Integer> groupsAddedToExchangeMetadatResponse = new HashSet<>();
+
+    MutableBoolean allReplicasCaughtUpEarly = new MutableBoolean(false);
+    Map<Integer, List<RemoteReplicaInfo>> groupIdToRemoteReplicaMap = new HashMap<>();
+
+    try {
+      storeKeyConverter.dropCache();
+      Map<DataNodeId, Integer> remoteHostToStandbyNoProgressReplicaGroupId = new HashMap<>();
+
+      generateGroupIdsForReplicas(groupIdToRemoteReplicaMap, remoteHostToStandbyNoProgressReplicaGroupId);
+
+      Map<Integer, RemoteReplicaGroup> inflightRemoteReplicaGroup = new HashMap<>();
+      Map<Integer, Integer> groupIdIterationCountMap = new HashMap<>();
+      Map<DataNodeId, Set<RemoteReplicaInfo>> dataNodeIdToPendingReplicasMarkedForStandByNoProgress = new HashMap<>();
+      Map<RemoteReplicaInfo, Long> remoteReplicaToThrottledTill = new HashMap<>();
+
+      Map<Integer, RemoteReplicaGroup> correlationIdToReplicaGroup = new HashMap<>();
+      Map<Integer, RequestInfo> correlationIdToRequestInfo = new LinkedHashMap<>();
+
+      MutableBoolean maxIterationsReached = new MutableBoolean(false);
+
+      do {
+        generateRemoteReplicaGroups(groupIdToRemoteReplicaMap, remoteHostToStandbyNoProgressReplicaGroupId,
```

Comment:
are there no more active/standby replicas ?

Reply:
there are active and standby replica groups, for both separate id is getting generated. For standby groups there is 1 id per data node to reduce number of cross colo calls, it is same as right now. Both will be added to inflight groups.